File: modules/cms/models/stock_movement.go
package models

import (
	"aldev/modules/global/models"

	"github.com/google/uuid"
)

type StockMovement struct {
	models.BaseModel
	ProductID     *uuid.UUID `json:"product_id" gorm:"type:uuid;not null"`
	Product       *Product   `json:"product,omitempty" gorm:"foreignKey:ProductID"`
	ReferenceType *string    `json:"reference_type" gorm:"type:varchar(50);not null"` // purchase, order, adjustment
	ReferenceID   *uuid.UUID `json:"reference_id" gorm:"type:uuid;not null"`
	Qty           *int       `json:"qty" gorm:"not null"` // positive for increment, negative for decrement
	Description   *string    `json:"description,omitempty" gorm:"type:text"`
}

File: src/types/stock-movement.ts
import type { Product } from './product';

export interface StockMovement {
   id?: string;
   created_at?: string;
   updated_at?: string;
   product_id?: string;
   product?: Product;
   reference_type?: 'purchase' | 'order' | 'adjustment';
   reference_id?: string;
   qty?: number; // positive for increment, negative for decrement
   description?: string;
}
